File: knowledge-base-frontend-master/src/services/notification.service.ts
import { http } from './request';
import { SystemNotification } from '@/types';

export const notificationService = {
  // Get notification list
  getNotifications: (params?: { page?: number; pageSize?: number; type?: string; isRead?: number }) => {
    const requestParams: Record<string, any> = {};
    if (params?.page !== undefined) {
      requestParams.current = params.page;
    }
    if (params?.pageSize !== undefined) {
      requestParams.size = params.pageSize;
    }
    if (params?.isRead !== undefined) {
      requestParams.isRead = params.isRead;
    }
    return http.get<{ records: SystemNotification[]; total: number; current: number; size: number }>('/notifications', {
      params: requestParams,
    });
  },

  // Get unread notification count
  getUnreadCount: () => {
    return http.get<number>('/notifications/unread-count');
  },

  // Mark as read
  markAsRead: (id: string) => {
    return http.put(`/notifications/${id}/read`);
  },

  // Mark all as read
  markAllAsRead: () => {
    return http.put('/notifications/read-all');
  },

  // Delete notification
  deleteNotification: (id: string) => {
    return http.delete(`/notifications/${id}`);
  },

  // Clear all notifications
  clearAll: () => {
    return http.delete('/notifications/all');
  },
};

export default notificationService;

File: knowledge-base-backend-master/kb-foundation/src/main/java/com/knowledge/base/foundation/service/NotificationService.java
package com.knowledge.base.foundation.service;

import com.baomidou.mybatisplus.core.metadata.IPage;
import com.knowledge.base.common.result.Result;
import com.knowledge.base.foundation.dto.NotificationDTO;
import com.knowledge.base.foundation.dto.NotificationQueryDTO;
import com.knowledge.base.foundation.entity.Notification;
import com.knowledge.base.foundation.vo.NotificationVO;

/**
 * Notification Service interface
 *
 * @author airwzz999
 * @since 1.0.0
 */
public interface NotificationService {

    /**
     * Send a notification (DTO-based)
     *
     * @param notificationDTO notification DTO
     * @return notification ID
     */
    Result<Long> sendNotification(NotificationDTO notificationDTO);

    /**
     * Paginated query of notifications
     *
     * @param current current page
     * @param size    page size
     * @param userId  user ID
     * @param isRead  whether read (optional)
     * @return paginated result
     */
    IPage<Notification> pageNotifications(Long current, Long size, Long userId, Integer isRead);

    /**
     * Query the notification list (query-DTO-based)
     *
     * @param queryDTO query conditions
     * @return paginated result
     */
    Result<IPage<NotificationVO>> getNotifications(NotificationQueryDTO queryDTO);

    /**
     * Get a notification by ID
     *
     * @param id notification ID
     * @return notification information
     */
    Notification getNotificationById(Long id);

    /**
     * Send a notification (entity-based, for internal calls)
     *
     * @param notification notification entity
     * @return whether it succeeded
     */
    Boolean sendNotification(Notification notification);

    /**
     * Mark a notification as read
     *
     * @param id notification ID
     * @return whether it succeeded
     */
    Result<Boolean> markAsRead(Long id);

    /**
     * Mark all notifications as read
     *
     * @param userId user ID
     * @return whether it succeeded
     */
    Result<Boolean> markAllAsRead(Long userId);

    /**
     * Delete a notification
     *
     * @param id notification ID
     * @return whether it succeeded
     */
    Result<Boolean> deleteNotification(Long id);

    /**
     * Get the unread notification count
     *
     * @param userId user ID
     * @return unread count
     */
    Result<Long> getUnreadCount(Long userId);
}
